coffee-chats/src/main/java/com/google/step/coffee/OAuthService.java

```diff
@@ -0,0 +1,113 @@
+package com.google.step.coffee;
+
+import static com.google.step.coffee.APIUtils.HTTP_TRANSPORT;
+import static com.google.step.coffee.APIUtils.JSON_FACTORY;
+import static com.google.step.coffee.data.CalendarUtils.SCOPES;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.extensions.appengine.datastore.AppEngineDataStoreFactory;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.auth.oauth2.GoogleTokenResponse;
+import com.google.api.client.http.GenericUrl;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import javax.servlet.http.HttpServletRequest;
+
+/** Provide tools for authorising permissions to Google APIs. */
+public class OAuthService {
+
+  private static final String CREDENTIALS_FILEPATH = "/client_secret.json";
+
+  private static GoogleClientSecrets clientSecrets = null;
+
+  /**
+   * Load credentials from store of pre-authorised user, null if user has not authorised.
+   *
+   * @param userId String of userId generated by AppEngine UserService.
+   * @return Credential object needed to interact with authorised Google APIs or null on fail.
+   */
+  public static Credential getCredentials(String userId) {
+    try {
+      GoogleAuthorizationCodeFlow flow = getAuthFlow();
+
+      return flow.loadCredential(userId);
+    } catch (IOException e) {
+      System.out.println("Getting credentials failed: " + e.getMessage());
```

Comment:
Is returning `null` an exceptional situation? If so, it might be better to just not catch the exception. If not, then printing it doesn't seem like a good idea.

Reply:
In situations where this is used, it would not make sense to attempt to handle an IOException. It makes more sense to handle this exception in a context where it still makes sense that the fetching of credentials is an I/O issue, rather than elsewhere where it should be opaque where the credentials are coming from.